compiler/rustc_infer/src/infer/error_reporting/need_type_info.rs

```diff
@@ -631,6 +626,24 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
             .any(|span_label| span_label.label.is_some() && span_label.span == span)
             && local_visitor.found_arg_pattern.is_none()
         {
+            let (kind_str, const_value) = match arg.unpack() {
+                GenericArgKind::Type(_) => ("type", None),
+                GenericArgKind::Const(_) => ("the value", Some(())),
+                GenericArgKind::Lifetime(_) => bug!("unexpected lifetime"),
+            };
+
+            if let Some(suggestion) = const_value
+                .and_then(|_| arg_data.parent_name.as_ref())
+                .and_then(|parent| Some(format!("{}::<{}>", parent, arg_data.name)))
+            {
+                err.span_suggestion_verbose(
+                    span,
+                    "consider specifying a const value in the function call",
```

Comment:
It'd be good to differentiate between function calls and method calls (this should be done automatically if you can fit it into the existing type error message handling below).
See https://github.com/rust-lang/rust/blob/e2de552f91df1668d4241b604ed7139216020d26/compiler/rustc_infer/src/infer/error_reporting/need_type_info.rs#L665-L668 and https://github.com/rust-lang/rust/blob/e2de552f91df1668d4241b604ed7139216020d26/compiler/rustc_trait_selection/src/traits/error_reporting/mod.rs#L1550-L1553
in particular.

Reply:
How would I check if it's a method call in this context?

Reply:
I'm not sure off the top of my head, because this code is a little convoluted. If you can't figure it out, just use:
```suggestion
                    "consider specifying the const argument",
```
and we can leave improving it for a follow up.